pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -257,6 +258,16 @@ Iterable<ArchiveIssue> syntaxCheckUrl(String url, String name) sync* {
   }
 }
 
+/// Validate that the package does not have too many dependencies.
+///
+/// It ignores `dev_dependencies` as these are for development only.
+Iterable<ArchiveIssue> validateDependencies(Pubspec pubspec) sync* {
+  if (pubspec.dependencies.length > 128) {
+    yield ArchiveIssue('Package must not exceed 128 direct dependencies. '
+        '(Please file an issue if you think you have a good reason for more dependencies.)');
```

Comment:
I just updated https://github.com/dart-lang/pub-dev/issues/3693 to say 100, can we do that?
Otherwise, this seems solid :D